stdlib/asm/collections/smt.masm

```diff
@@ -301,3 +318,393 @@ export.get
     end
     # => [V, R, ...]
 end
+
+# INSERT
+# =================================================================================================
+
+#! Updates a leaf node at depths 16, 32, or 48.
+#!
+#! Input:  [d, idx, V, K, R, ...];
+#! Output: [V_old, R_new, ...]
```

Comment:
Yes, I was trying to use it as well, but found it difficult because sometimes I would use "3 states" - e.g., `P`, `P_new`, and `P_old`, where `P` is an expected value of the node which should be equal to `P_old`.